src/main/app/src/components/NavBar.js

```diff
@@ -121,6 +124,54 @@ const ContentSwitch = () => {
   );
 };
 
+const LogOutButton = () => {
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const LogOutDialogBox = () => {
```

Comment:
Also looks good, but would be better refactored into a separate component function.

If you need to pass in `LogOutButton` internal state to the `LogOutDialogBox` component, then do it with props:

```javascript
<LogOutDialogBox open={open} onClose={handleClose} />
```